Various docstring fixes

ChiefMilesEdgeworth opened the pull request:
- [x] xref #27976
- [x] tests added / passed - Didn't run all tests, only those concerning the issue (`./scripts/validate_docstrings.py pandas.Series.drop --errors=PR02`)

- [x] passes `black pandas` - Only docstrings were edited, so there were no major changes 
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] ~~whatsnew entry~~ -  Not needed, only docs changed
General fixes to the documentation in different places. I discovered that the root of many of the conflicts lies in decorators. We might need to re-evaluate how we're getting the signature of functions in the validation script. 

First time submitting a PR! Let me know if I need to do anything, or if there's anything I should do differently next time :)

ChiefMilesEdgeworth committed:
DOC: Fixed PR02 docstring error in pandas Series idxmin and idxmax

Simple error fixes for idxmin and idxmax regarding their docstrings

ChiefMilesEdgeworth committed:
DOC: Fixed other PR02 docstring errors in pandas Series class

Fixed errors on swaplevel, rename, and drop methods

ChiefMilesEdgeworth committed:
DOC: Fixed other PR02 docstring errors in pandas rolling and ewm files

Fixed methods mean, std, var, and corr in ewm.py. 
Fixed methods sum, apply, max, std, var, and quantile in rolling.py

ChiefMilesEdgeworth committed:
DOC: Fixed other PR02 docstring errors in pandas plotting file

Fixed functions table, scatter_matrix, radviz, bootstrap_plot, parallel_coordinates, lag_plot, and autocorrelation_plot. Noticed that functions with the @deprecate_kwarg decorator tend to not behave correctly with the tests.

ChiefMilesEdgeworth committed:
DOC: Fixed other PR02 docstring errors in pandas frame.py file

Fixed the docstring for transpose, eval, swaplevel, and melt in frame.py

ChiefMilesEdgeworth committed:
Ran Black

Just missed the one line.

ChiefMilesEdgeworth committed:
Flake8 Changes

Caught small whitespace issues.

ChiefMilesEdgeworth committed:
Merge branch 'master' into Various_docstring_fixes

ChiefMilesEdgeworth committed:
Test fixes

Fixed errors shown by automated tests (PR04 and GL03)

angelaambroz commented:
Oh, uh - I guess this PR re-does what [mine did](https://github.com/pandas-dev/pandas/pull/28742) - plus additional stuff. No sense merging both, so looks like this one might be a superset of mine. So much for my first PR on pandas... 😞

Nabel0721 commented:
Sorry, I didn't see yours until after I had committed the overlapping
changes. There's still more than a few that look like simple formatting
errors if you want to go after those. I'm going to start looking into the
issues with decorators.
Don't feel disheartened, my work wasn't any more valuable than yours.

On Wed, Oct 2, 2019 at 11:23 AM Angela Ambroz <notifications@github.com>
wrote:

> Oh, uh - I guess this PR re-does what mine did
> <https://github.com/pandas-dev/pandas/pull/28742> - plus additional
> stuff. No sense merging both, so looks like this one might be a superset of
> mine. So much for my first PR on pandas... 😞
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/pandas-dev/pandas/pull/28744?email_source=notifications&email_token=ACT76UTP2DHUISSB4XZSH7LQMS4HVA5CNFSM4I4RUVZKYY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOEAFEO3I#issuecomment-537544557>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ACT76US5D7W4EIHUZ5TPTU3QMS4HVANCNFSM4I4RUVZA>
> .
>


-- 
Sincerely,

*Nathan Abel*
Kettering University - Computer Science
989-780-3050

ChiefMilesEdgeworth committed:
DOC: Test fixes part 2

Fixed issues shown by automated tests with error PR03.

angelaambroz commented:
Thanks for the follow-up. Yeah, it's hard to find unclaimed work on these projects. (I've been looking here and on scikit-learn.) I'll take a look at the remaining stuff. 👍

ChiefMilesEdgeworth committed:
Merge branch 'master' into Various_docstring_fixes

Nabel0721 commented:
I'm not sure why the Travis CI build failed. Could I get a pointer on how to fix it, or was the code not at fault?

datapythonista commented on pandas/core/frame.py:
```diff
@@ -2610,10 +2610,10 @@ def transpose(self, *args, **kwargs):
+        *args
+            Additional arguments have no effect but might be accepted for
+            compatibility with numpy.
+        **kwargs
```
The script is wrong in this case, please revert

datapythonista commented on pandas/core/frame.py:
```diff
@@ -3237,7 +3237,7 @@ def eval(self, expr, inplace=False, **kwargs):
             operation inplace and mutate the existing DataFrame. Otherwise,
             a new DataFrame is returned.
-        kwargs : dict
+        **kwargs : dict
```
please remove the type

datapythonista commented on pandas/core/series.py:
```diff
@@ -2076,13 +2076,16 @@ def idxmin(self, axis=0, skipna=True, *args, **kwargs):
+        *args
+            Additional arguments have no effect but might be accepted
+            for compatability with NumPy
+        **kwargs
```
merge those please

datapythonista commented on pandas/core/frame.py:
```diff
@@ -5205,8 +5205,10 @@ def swaplevel(self, i=-2, j=-1, axis=0):
         ----------
-        i, j : int, str (can be mixed)
-            Level of index to be swapped. Can pass level name as string.
+        i : int, str (can be mixed)
```
`int or str`, not sure what the can be mixed means here, but shouldn't be in the type (can be explained in the description)

datapythonista commented on pandas/core/series.py:
```diff
@@ -2146,13 +2149,16 @@ def idxmax(self, axis=0, skipna=True, *args, **kwargs):
+        *args
+            Additional arguments have no effect but might be accepted
+            for compatibility with NumPy.
+        **kwargs
```
please merge args and kwargs

datapythonista commented on pandas/core/series.py:
```diff
@@ -3525,8 +3531,10 @@ def swaplevel(self, i=-2, j=-1, copy=True):
-            Level of index to be swapped. Can pass level name as string.
+        i : int, str
+            Level of first index to be swapped. Can pass level name as string.
+        j : int, str
```
same as before

datapythonista replied:
same

datapythonista commented on pandas/core/series.py:
```diff
@@ -4165,6 +4169,10 @@ def drop(
             No change is made to the Series; use 'index' or 'labels' instead.
 
+            .. versionadded:: 0.21.0
+        columns : None
```
this is repeated, what's the reason for this?

datapythonista commented on pandas/core/window/ewm.py:
```diff
@@ -17,27 +17,10 @@
+            Arguments to be passed into func.
         **kwargs
-           Keyword arguments to be passed into func.
+            Keyword arguments to be passed into func.
```
merge args and kwargs

datapythonista commented on pandas/core/window/ewm.py:
```diff
@@ -348,11 +349,26 @@ def _get_cov(X, Y):
+            output will be a MultiIndex DataFrame in the case of DataFrame
+            inputs. In the case of missing elements, only complete pairwise
+            observations will be used.
+        **kwargs : dict of {str : Any}
```
remove the type

datapythonista commented on pandas/core/window/ewm.py:
```diff
@@ -348,11 +349,26 @@ def _get_cov(X, Y):
+        **kwargs : dict of {str : Any}
+           Keyword arguments to be passed into func.
         """
+
```
remove this blank line, this is probably breaking the CI

datapythonista commented on pandas/core/window/rolling.py:
```diff
@@ -1018,7 +1021,9 @@ def count(self):
-    *args, **kwargs
+    args : tuple of Any
+        Arguments and keyword arguments to be passed into func.
+    kwargs : dict of {str : any}
```
remove the types, aren't those `*args` instead of `args`?

datapythonista commented on pandas/plotting/_misc.py:
```diff
@@ -398,7 +398,7 @@ def lag_plot(series, lag=1, ax=None, **kwds):
     lag : lag of the scatter plot, default 1
     ax : Matplotlib axis object, optional
-    kwds : Matplotlib scatter method keyword arguments, optional
+    **kwds : Matplotlib scatter method keyword arguments, optional
```
move this to the description, no type for star params

datapythonista reviewed (commented):
Thanks a lot for working on all those. Added some comments, all the args and kwargs should be merged, unless they really require different descriptions.

ChiefMilesEdgeworth committed:
DOC: Update args and kwargs to same line, other small changes

Put args and kwargs back on the same line, since that's how the documentation should be. The validation script needs to be changed to catch this.

ChiefMilesEdgeworth committed:
Merge remote-tracking branch 'origin/Various_docstring_fixes' into Various_docstring_fixes

pep8speaks commented:
Hello @Nabel0721! Thanks for updating this PR. We checked the lines you've touched for [PEP 8](https://www.python.org/dev/peps/pep-0008) issues, and found:

* In the file [`pandas/core/frame.py`](https://github.com/pandas-dev/pandas/blob/2f48f5ad8654da38b220ade134facda8cd5c9730/pandas/core/frame.py):

> [Line 2614:74](https://github.com/pandas-dev/pandas/blob/2f48f5ad8654da38b220ade134facda8cd5c9730/pandas/core/frame.py#L2614): [W291](https://duckduckgo.com/?q=pep8%20W291) trailing whitespace

* In the file [`pandas/core/series.py`](https://github.com/pandas-dev/pandas/blob/2f48f5ad8654da38b220ade134facda8cd5c9730/pandas/core/series.py):

> [Line 2158:1](https://github.com/pandas-dev/pandas/blob/2f48f5ad8654da38b220ade134facda8cd5c9730/pandas/core/series.py#L2158): [W293](https://duckduckgo.com/?q=pep8%20W293) blank line contains whitespace

ChiefMilesEdgeworth committed:
DOC: Missed whitespace

Should've checked it before commiting. Two very small fixes

Nabel0721 commented:
I'm going to start looking into fixing the validation script so it catches more of these properly, since the correct format for the docs is to have args and kwargs in the same line.

datapythonista commented:
Can't look now, but I think there is an issue for it.

Nabel0721 commented:
I think this (#20298) is it. If you'd like I can comment on there with this issue. There's another part I'd like to fix as well, which I commented about in #27976.

datapythonista commented:
Sure, will be great to see both fixed.

Nabel0721 commented:
This is ready to be merged, I have the changes to the validation script in a separate pull request, since it addresses a different problem.

datapythonista commented on pandas/core/window/ewm.py:
```diff
@@ -275,8 +256,10 @@ def mean(self, *args, **kwargs):
+        *args
+            Arguments to be passed into func.
+        **kwargs
+            Keyword arguments to be passed into func.
```
Better revert this please.

datapythonista replied:
Any reason to not resolve this and mark as resolved? I think your proposed change is incorrect, the original is more accurate.

Nabel0721 replied:
No idea why I missed these. I'll get them right away.

Nabel0721 replied:
Actually, it looks like I just forgot to mark this as resolved. It should be fixed in the last commit.

datapythonista replied:
I think it's not reverted.

datapythonista replied:
Can you revert please

Nabel0721 replied:
For some reason it's showing up as reverted on my machine, but not here. I'll just do the edits on Github.

datapythonista commented on pandas/core/series.py:
```diff
@@ -4165,6 +4163,10 @@ def drop(
 
+            .. versionadded:: 0.21.0
+        columns : None
+            Redundant for application on Series.
```
Columns was already documented, remove the repeated one

datapythonista commented on pandas/plotting/_misc.py:
```diff
@@ -168,7 +168,7 @@ def radviz(frame, class_column, ax=None, color=None, colormap=None, **kwds):
         Colormap to select colors from. If string, load colormap with that
         name from matplotlib.
-    kwds : optional
+    **kwds : optional
```
remove the optional

datapythonista reviewed (commented):
Some comments, other than that looks good

ChiefMilesEdgeworth committed:
DOC: Fixing overlooked issues

Merged *args and **kwargs, removed a duplicate documentation, and removed the type for **kwargs.

Nabel0721 commented:
Issues are fixed. Thanks for working through this with me!

datapythonista commented on pandas/core/series.py:
```diff
@@ -3525,8 +3525,10 @@ def swaplevel(self, i=-2, j=-1, copy=True):
-            Level of index to be swapped. Can pass level name as string.
+        i : int, str
+            Level of first index to be swapped. Can pass level name as string.
+        j : int, str
```
`int or str` in both cases

datapythonista replied:
and you can also merge them back

datapythonista commented on pandas/core/frame.py:
```diff
@@ -5205,8 +5202,10 @@ def swaplevel(self, i=-2, j=-1, axis=0):
+        i : int or str
+            Level of first index to be swapped. Can pass level name as string.
+        j : int or str
+            Level of second index to be swapped. Can pass level name as string.
```
I'd merge those, I think the validation script should accept this, as proposed in the PR where you update it.

Nabel0721 replied:
Right now the proposed script does not validate any variables on the same line, just *args and **kwargs. I left it that way in case we decide we don't want people to just throw a bunch of variables on the same line. If you think that'd be fine, then I can rewrite the script update to allow for that, and change these here.

datapythonista replied:
I think the PR allowing having those together was merged, right? Can you have `i, j` instead please

datapythonista reviewed (commented):
couple of small things, but looks good besides that

Nabel0721 commented:
Just saw that you also looked at the other PR and requested that it was split on "," there. I'll work on that and get these fixed back to how they were before.

ChiefMilesEdgeworth committed:
Merge pull request #1 from pandas-dev/master

Update Local Version

ChiefMilesEdgeworth committed:
Merge remote-tracking branch 'origin/master'

ChiefMilesEdgeworth committed:
Merge remote-tracking branch 'upstream/master'

WillAyd commented:
@Nabel0721 can you merge master for this now the other change is in?

Nabel0721 commented:
Yes, it says there are no conflicts. Would you like me to rerun the checks? I'm not sure how I'd do that.

WillAyd commented:
If you can merge master and repush would be ideal

ChiefMilesEdgeworth committed:
Merge remote-tracking branch 'upstream/master'

Updating to most recent version

ChiefMilesEdgeworth committed:
Merge branch 'master' into Various_docstring_fixes

datapythonista commented on pandas/plotting/_misc.py:
```diff
@@ -398,7 +398,8 @@ def lag_plot(series, lag=1, ax=None, **kwds):
     ax : Matplotlib axis object, optional
-    kwds : Matplotlib scatter method keyword arguments, optional
+    **kwds
+        Matplotlib scatter method keyword arguments, optional
```
Can you remove the optional, and end with period.

ChiefMilesEdgeworth committed:
DOC: Revert back to multiple parameters on one line

Moved a couple of index parameters onto the same line, and redid one parameter description

ChiefMilesEdgeworth committed:
Merge remote-tracking branch 'upstream/master' into Various_docstring_fixes

ChiefMilesEdgeworth committed:
Merge remote-tracking branch 'upstream/master' into Various_docstring_fixes

datapythonista commented on pandas/core/frame.py:
```diff
@@ -5209,8 +5206,8 @@ def swaplevel(self, i=-2, j=-1, axis=0):
 
         Parameters
         ----------
-        i, j : int, str (can be mixed)
```
restore the space before the colon

datapythonista commented on pandas/core/frame.py:
```diff
@@ -5209,8 +5206,8 @@ def swaplevel(self, i=-2, j=-1, axis=0):
-        i, j : int, str (can be mixed)
-            Level of index to be swapped. Can pass level name as string.
+        i, j: int or str
+            Levels indecies to be swapped. Can pass level name as string.
```
typo, indices

datapythonista commented on pandas/core/series.py:
```diff
@@ -3544,8 +3544,8 @@ def swaplevel(self, i=-2, j=-1, copy=True):
-        i, j : int, str (can be mixed)
-            Level of index to be swapped. Can pass level name as string.
+        i, j : int, str
+            Level of the indecies to be swapped. Can pass level name as string.
```
same typo

datapythonista reviewed (changes requested):
Nice fixes, few comments

ChiefMilesEdgeworth committed:
DOC: Spacing fixed

Added a space between a parameter and the colon.

ChiefMilesEdgeworth committed:
DOC: Fixed typos

Fixed typos on documentation

datapythonista commented on pandas/core/frame.py:
```diff
@@ -5209,8 +5206,8 @@ def swaplevel(self, i=-2, j=-1, axis=0):
-        i, j : int, str (can be mixed)
-            Level of index to be swapped. Can pass level name as string.
+        i, j : int or str
+            Levels indices to be swapped. Can pass level name as string.
```
```suggestion
            Levels of the indices to be swapped. Can pass level name as string.
```
Clearer and consistent with the other docstrings.

datapythonista commented on pandas/core/window/ewm.py:
```diff
@@ -276,7 +257,7 @@ def mean(self, *args, **kwargs):
         ----------
         *args, **kwargs
-            Arguments and keyword arguments to be passed into func.
+            Keyword arguments to be passed into func.
```
```suggestion
            Arguments and keyword arguments to be passed into func.
```
You can commit this from the UI.

datapythonista reviewed (commented):
Almost there, couple of comments.

datapythonista commented:
@Nabel0721 do you have time to commit the last changes?

And next time it'd be better if you leave checked the checkbox to let maintainers commit to your branch, and we could do it ourselves.

Thanks!

Nabel0721 commented:
I got really busy unexpectedly, so I don't think I'll have time for a couple more weeks. I do think I have the "allow edits from maintainers" checkbox on, so I'm not sure why it's showing up as off on your end. If you are able to make edits though, feel free.

WillAyd commented:
Nice PR but sounds like it is stale (or at least will be for the foreseeable future).

@Nabel0721 ping if you'd like to pick this back up and can reopen

Nabel0721 commented:
Yeah, my apologies. I’m not going to be able to get to this for a while.
I’ll be sure to ping if there’s an update.

On Thu, Nov 7, 2019 at 7:05 PM William Ayd <notifications@github.com> wrote:

> Closed #28744 <https://github.com/pandas-dev/pandas/pull/28744>.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/pandas-dev/pandas/pull/28744?email_source=notifications&email_token=ACT76UWBBBVR7LTRSKCNXVDQSSUN5A5CNFSM4I4RUVZKYY3PNVWWK3TUL52HS4DFWZEXG43VMVCXMZLOORHG65DJMZUWGYLUNFXW5KTDN5WW2ZLOORPWSZGOUXDNNGY#event-2781271707>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ACT76UTH7RIBWJPLBE77W6TQSSUN5ANCNFSM4I4RUVZA>
> .
>
-- 
Sincerely,

*Nathan Abel*
Kettering University - Computer Science
989-780-3050

datapythonista commented:
@Nabel0721 the only thing pending is literally clicking on the `Commit suggestions` buttons in this comment: https://github.com/pandas-dev/pandas/pull/28744#pullrequestreview-302294656

We can't do it ourselves (github issue, or you didn't allow maintainers to edit this PR). If you can spend one minute doing that, we can merge this changes.

ChiefMilesEdgeworth committed:
Update pandas/core/frame.py

I didn't know I could do this

Co-Authored-By: Marc Garcia <garcia.marc@gmail.com>

ChiefMilesEdgeworth committed:
Update pandas/core/window/ewm.py

Small change to wording

Co-Authored-By: Marc Garcia <garcia.marc@gmail.com>

Nabel0721 commented:
Sorry, I didn't know I could do that. I would have done it a while ago if I had known.

Nabel0721 commented:
It must be a github issue that you can't do edits though. I checked again and the allow edits checkbox is still checked. Sorry for how long this has taken to finish up, but thank you for being patient.

datapythonista reviewed (approved):
lgtm, thanks a lot @Nabel0721 for fixing all this. Hope to see you back when you have more free time!

Nabel0721 commented:
For sure. It's always nice to get to work on python code.

WillAyd commented:
Great thanks @Nabel0721 and @datapythonista
